### DISCUSSION TITLE
modal is blocked when used in dockview

### DISCUSSION THREAD

**Discussion by celadaris** [dotnetcore/BootstrapBlazor#4266] [Q&A] [answered]
### Is there an existing issue for this?

- [x] I have searched the existing issues

### Is your feature request related to a problem? Please describe the problem.

i am trying to use a modal (the modal is within BootsrapBlazor.Components.Modal) in a component within a component, so the modal is 2 layers deep. Because of this the modal starts to experience issues i think because its too deep.

using the BlazorPortals project fixes the modal by rendering it to a parent component, but that is a depricated project...

### Describe the solution you'd like

because of these issues it would be really cool to see some kind of a portal implementation in BootstrapBlazor. But while there isnt one ill be using the depricated library for now...

### Additional context

I can post code of an example of putting a modal 2 child components deep, but it should be easy to reproduce...

**Comment by bb-auto** [comment]
@celadaris Thank you for contacting us. We will give feedback later.

**Comment by ArgoZhang** [comment]
@celadaris hi thanks for you contacting us. please submit the mini-repo-project for this issue. let us to known the `BlazorPortals`

**Comment by celadaris** [comment]
My apologies, after doing extensive amounts of testing i found that my issue only appears when im using DockView version 8.1.1 when upgrading to DockView version 8.1.3 it seems to fix the bug.

i kind of mislead you into thinking it happens when the modal is a child of a child and its two layers deep but that was an incorrect cause of the bug, the bug actually happens when you launch a modal inside a DockViewV2

because the bug is fixed when upgrading to the latest version
 of DockView i will close this, but ill still post my repo project if your curious.

this repo displays the bug
[BrokenModal.zip](https://github.com/user-attachments/files/16922968/BrokenModal.zip)

this repo is a fix using portals (not recommended)
[BrokenModal.zip](https://github.com/user-attachments/files/16923042/BrokenModal.zip)

**Comment by celadaris** [comment]
hate to reopen this, but... when popping out the DockViewComponent and attempting to open the modal, the bug appears, I have uploaded two new repo's they are slightly modified from the last ones.

this repo displays the bug
[BrokenModal.zip](https://github.com/user-attachments/files/16950064/BrokenModal.zip)

this repo is a fix using portals (not recommended)
[BrokenModal.zip](https://github.com/user-attachments/files/16950085/BrokenModal.zip)

https://github.com/user-attachments/assets/6755f547-b73f-4b82-9bb4-c6a5df46a826

**Comment by ArgoZhang** [comment]
@celadaris got. let us resolve this issue. thank you report this

**Comment by ArgoZhang** [comment]
@celadaris 


https://github.com/user-attachments/assets/eb0834a7-5565-41af-9f55-6d99a46dac1f

use `DialogService` to replace `Modal`

please download this demo and try it. please let us know if you have any questions.

[BrokenModal.zip](https://github.com/user-attachments/files/16956462/BrokenModal.zip)

**Comment by celadaris** [comment]
It's not clear how one can use DialogService in a way that avoids having to deal with reading code as a string, and its unclear if it's possible to use advanced features like javascript interop/blazorSSR. As an alternative I've found out that blazor natively supports the idea of a Portal, but its called [Section](https://learn.microsoft.com/en-us/aspnet/core/blazor/components/sections?view=aspnetcore-8.0) I think ill stick to this because I wont have to depend on deprecated libraries, ill have clean code, and creating modals with advanced features and a lot of html is the same workflow as anything else.

this repo is a fix using `Section`
[BrokenModal.zip](https://github.com/user-attachments/files/16964008/BrokenModal.zip)

**Comment by ArgoZhang** [comment, marked as answer]
`DialogService` is a very powerful component and is a must-know point.

you can get some information from 

https://www.blazor.zone/dialog-service
https://www.blazor.zone/dialog

`SectionContent` is an alternative way. Modal and Dialog have different design purposes. Modal is rendered when the page is rendered. Dialog is rendered dynamically.

`DialogOption.BodyTemplate` and `DialogOption.Component` two parameters can adapt to all situations. You can dynamically render any component using `BootstrapDynamicComponent.CreateComponent` static method

**Reply by celadaris** [reply]
i was running some heavy queries on these modal's, switching to dialog has helped so much! THANK YOU! 😆

**Reply by ArgoZhang** [reply]
@you are welcome. It is strongly recommended to refactor to use Dialog service, which has better performance and less code.
